### PR TITLE
[CALCITE] Added support for CREATE TABLE attribute DATABLOCKSIZE

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 1]
```diff
@@ -236,6 +236,35 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
...
+        )
+        <DATABLOCKSIZE> { unitSize = DataBlockUnitSize.UNSPECIFIED; }
+    |
+        <DATABLOCKSIZE> <EQ> dataBlockSize = UnsignedNumericLiteral()
+        (
+            <BYTES> { unitSize = DataBlockUnitSize.BYTES; }
+        |
+            ( <KILOBYTES> | <KBYTES> ) { unitSize = DataBlockUnitSize.KILOBYTES; }
+        |
+            { unitSize = DataBlockUnitSize.UNSPECIFIED; }
```
Comment (AndrewPochapsky) [inline babel/src/main/codegen/includes/parserImpls.ftl:262]: Should this be DataBlocUnitSize.BYTES? I saw in the DataBlockUnitSize enum definition that the comment for DataBlockUnitSize.UNSPECIFICIED says the default size is BYTES. If so, wouldn't it make sense to have BYTES be the default if the user doesn't specify a unit?

Comment (michaelvdow) [reply babel/src/main/codegen/includes/parserImpls.ftl:262]: That's a good point. If a user doesn't assign a unit size we could just default to BYTES, and appending BYTES when unparsing would just give the user an equivalent statement

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -236,6 +236,33 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
...
+SqlCreateAttribute CreateTableAttributeDataBlockSize() :
+{
+    DataBlockModifier modifier = null;
+    DataBlockUnitSize unitSize;
+    SqlLiteral dataBlockSize = null;
+}
+{
+    (
+        (
+            ( <MINIMUM> | <MIN> ) { modifier = DataBlockModifier.MINIMUM; }
```
Comment (dasch-1) [inline babel/src/main/codegen/includes/parserImpls.ftl:248]: Nice, I like this pattern.

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 3]
```diff
@@ -340,6 +340,60 @@ private void checkParseInfixCast(String sqlType) {
...
+    final String expected = "CREATE TABLE `FOO`, DATABLOCKSIZE = 42.123 KILOBYTES (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeDataBlockSizeValueKilobytes() {
+    final String sql = "create table foo, datablocksize = 2e4 kilobytes (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, DATABLOCKSIZE = 2E4 KILOBYTES (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
```
Comment (dstekol) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:396]: Could also add a test case checking that having both a keyword before DATABLOCKSIZE and an assignment after (ex "MAX DATABLOCKSIZE = 42" ) fails.

Comment (dasch-1) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:396]: +1

Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:396]: Good point, I'll add a test case for that

**Pull request merged**